noxfile.py

```diff
@@ -0,0 +1,173 @@
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def format(session):
+    """Format the code."""
+    session.install(*REQUIREMENTS_FORMAT)
+    # TODO: incremental mode for yapf
+    session.run('yapf', '--in-place', '--parallel', '--recursive', *PY_PATHS)
+    # TODO: uncomment if we want to use isort and docformatter
+    # session.run('isort', *PY_PATHS)
+    # session.run(
+    #     'docformatter',
+    #     '--in-place',
+    #     '--recursive',
+    #     '--wrap-summaries=100',
+    #     '--wrap-descriptions=100',
+    #     *PY_PATHS,
+    # )
+
+
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def lint(session):
+    """Run linters."""
+    session.install(*REQUIREMENTS_LINT)
+    session.run('yapf', '--diff', '--parallel', '--recursive', *PY_PATHS)
+    # TODO: uncomment if we want to use isort and docformatter
+    # session.run('isort', '--check', *PY_PATHS)
+    # session.run(
+    #     'docformatter',
+    #     '--check',
+    #     '--recursive',
+    #     '--wrap-summaries=100',
+    #     '--wrap-descriptions=100',
+    #     *PY_PATHS,
+    # )
+
+    # TODO: use flake8 instead of pyflakes
+    session.log('pyflakes b2')
+    output = subprocess.run('pyflakes b2', shell=True, check=False,
+                            stdout=subprocess.PIPE).stdout.decode().strip()
+    excludes = ['__init__.py']
+    output = [l for l in output.splitlines() if all(x not in l for x in excludes)]
+    if output:
+        print('\n'.join(output))
+        session.error('pyflakes has failed')
+    # session.run('flake8', *PY_PATHS)
+    session.run('pytest', 'test/static')
+
+
+@nox.session(python=PYTHON_VERSIONS)
+def unit(session):
+    """Run unit tests."""
+    install_myself(session)
+    session.install(*REQUIREMENTS_TEST)
+    session.run(
+        'pytest', '--cov=b2', '--cov-branch', '--cov-report=xml', '--doctest-modules',
+        *session.posargs, 'test/unit'
+    )
+    session.notify('cover')
+
+
+@nox.session(python=PYTHON_VERSIONS)
+def integration(session):
+    """Run integration tests."""
+    install_myself(session)
+    session.install(*REQUIREMENTS_TEST)
+    session.run('pytest', '-s', *session.posargs, 'test/integration')
+
+
+@nox.session(python=PYTHON_VERSIONS)
+def test(session):
+    """Run all tests."""
+    if session.python:
+        session.notify('unit-{}'.format(session.python))
+        session.notify('integration-{}'.format(session.python))
+    else:
+        session.notify('unit')
+        session.notify('integration')
+
+
+@nox.session
+def cover(session):
+    """Perform coverage analysis."""
+    session.install('coverage')
+    session.run('coverage', 'report', '--fail-under=75', '--show-missing')
+    session.run('coverage', 'erase')
+
+
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def build(session):
+    """Build the distribution."""
+    # TODO: consider using wheel as well
+    install_myself(session)
+    session.install(*REQUIREMENTS_BUILD)
+    session.run('liccheck', '-s', 'setup.cfg')
```

Comment:
so we only run it when trying to build a package? Someone adding a new dependency will have a very unpleasant surprise very late in the process. Can we move it up somewhere where it will provide feedback faster?

Reply:
Honestly, I had a problem where should I put it. At the beginning I added it to the `lint` session, but it should be fast and we don't add new requirements very often, so why slowing down? Any idea?

Reply:
isn't it blazingly fast?

I think you can put it after the unit tests when run from the console, then it doesn't inhibit negative feedback in TDD workflow

Reply:
It is fast, however, it requires that all dependencies are installed and currently in `lint` session we don't need it, but `nox reuse virtualenv` feature kind of solve it, so it's not a that big problem.
The `unit` session doesn't fit at all because it supports multiple python interpreters and it doesn't make sense for license checking.
I think I will move it to `lint`

Reply:
maybe the default it should be lint, test, liccheck?

Reply:
I've moved liccheck to the `lint` session